account_invoice_section_sale_order/models/sale_order.py

```diff
@@ -55,6 +55,8 @@ def _get_saleorder_section_name(self):
         """Returns the text for the section name."""
         self.ensure_one()
         if self.client_order_ref:
-            return "{} - {}".format(self.name, self.client_order_ref or "")
+            return "{} - {} - {}".format(
+                self.name, self.client_order_ref or "", self.partner_id.name
```

Comment:
That could be a configured because maybe not everyone wants to have partner name in the section